examples/simulation/plot_simulated_sample_dataset.py

```diff
@@ -0,0 +1,182 @@
+data_path = sample.data_path()
+subjects_dir = op.join(data_path, 'subjects')
+subject = 'sample'
+
+# First, we get an info structure from the sample subject.
+evoked_fname = op.join(data_path, 'MEG', subject, 'sample_audvis-ave.fif')
+info = mne.io.read_info(evoked_fname)
+tstep = 1 / info['sfreq']
+
+# To simulate sources, we also need a source space. It can be obtained from the
+# forward solution of the sample subject.
+meg_path = op.join(data_path, 'MEG', subject)
+fwd_fname = op.join(meg_path, 'sample_audvis-meg-eeg-oct-6-fwd.fif')
+fwd = mne.read_forward_solution(fwd_fname)
+src = fwd['src']
+
+
+################################################################################
+
+# Load the real raw data and corresponding events.
+raw = mne.io.read_raw_fif(op.join(meg_path, 'sample_audvis_raw.fif'))
+raw.set_eeg_reference(projection=True).crop(0, 60)  # for speed
+
+# Standard sample event IDs. These values will correspond to the third column
+# in the events matrix.
+event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
+            'visual/right': 4, 'smiley': 5, 'button': 32}
+# Events from the experiment.
+events = mne.find_events(raw)
+
+# ##############################################################################
+
+# In order to simulate source time courses, labels of desired active regions
+# need to be specified for each of the 4 simulation conditions.
+# Make a dictionary that maps conditions to activation strengths within
+# aparc.a2009s labels. In the aparc.a2009s parcellation:
+#   -  'G_temp_sup-G_T_transv' is the label for primary auditory area 
+#   -  'S_calcarine' is the label for primary visual area
+# In each of the 4 conditions, only the primary area is activated. This means
+# that during the activations of auditory areas, there are no activations in
+# visual areas and vice versa.
+# Moreover, for each condition, contralateral region is more active (here, 2
+# times more) than the ipsilateral.
+
+activations = {
+    'auditory/left':
+        [('G_temp_sup-G_T_transv-lh', 100),          # label, activation (nAm)
+         ('G_temp_sup-G_T_transv-rh', 200)],
+    'auditory/right':
+        [('G_temp_sup-G_T_transv-lh', 200),
+         ('G_temp_sup-G_T_transv-rh', 100)],
+    'visual/left':
+        [('S_calcarine-lh', 100),
+         ('S_calcarine-rh', 200)],
+    'visual/right':
+        [('S_calcarine-lh', 200),
+         ('S_calcarine-rh', 100)],
+}
+
+annot='aparc.a2009s'
+
+# Load the 4 necessary label names.
+label_names = sorted(set(activation[0]
+                         for activation_list in activations.values()
+                         for activation in activation_list))
+region_names = list(activations.keys())
+
+#  Define the time course of the activity for each region to activate. We use a
+#  sine wave and it will be the same for all 4 regions.
+source_time_series = np.sin(np.linspace(0, 4 * np.pi, 100)) * 10e-9
+
+
+# Create simulated source activity.
```

Comment:
Could use another line of `#############` above this, and maybe an underline of `-------` to make this a proper title. Comments in blocks that start with `########` end up being rendered in our docs as proper restructuredtext so we should take advantage of that to give more structure here